src/main/java/nextstep/member/application/TokenService.java

```diff
@@ -2,24 +2,39 @@
 
 import lombok.RequiredArgsConstructor;
 import nextstep.member.AuthenticationException;
-import nextstep.member.presentation.TokenResponse;
 import nextstep.member.domain.Member;
+import nextstep.member.infrastructure.MemberRepository;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+@Transactional(readOnly = true)
 @RequiredArgsConstructor
 @Service
 public class TokenService {
-    private final MemberService memberService;
+    private final MemberRepository memberRepository;
     private final JwtTokenProvider jwtTokenProvider;
+    private final GithubClient githubClient;
 
     public TokenResponse createToken(String email, String password) {
-        Member member = memberService.findMemberByEmail(email);
+        var member = memberRepository.findByEmail(email).orElseThrow(RuntimeException::new);
         if (!member.getPassword().equals(password)) {
             throw new AuthenticationException();
         }
 
-        String token = jwtTokenProvider.createToken(member.getEmail());
+        var token = jwtTokenProvider.createToken(member.getEmail());
 
         return new TokenResponse(token);
     }
+
+    @Transactional
+    public TokenResponse getAuthToken(final String code) {
+        var tokenResponse = githubClient.getAccessTokenFromGithub(code);
+        var githubProfileResponse = githubClient.requestGithubProfile(tokenResponse.getAccessToken());
+
+        var email = githubProfileResponse.getEmail();
+        memberRepository.findByEmail(email)
+                .orElseGet(() -> memberRepository.save(new Member(email)));
+
+        return new TokenResponse(jwtTokenProvider.createToken(email));
```

Comment:
```suggestion
        var member = memberRepository.findByEmail(email)
                .orElseGet(() -> memberRepository.save(new Member(email)));

        return new TokenResponse(jwtTokenProvider.createToken(member.getEmail()));
```

memberRepository로 정확히 찾은 이메일로 다음 행위를 하는 게 좋지 않을까 하네요~